packages/cli/src/commands/init/init.ts

```diff
@@ -17,14 +17,18 @@ import {
   copyTemplate,
   executePostInitScript,
 } from './template';
-import {changePlaceholderInTemplate} from './editTemplate';
+import {
+  changePlaceholderInTemplate,
+  changeReactNativeVersionInTemplate,
+} from './editTemplate';
 import * as PackageManager from '../../tools/packageManager';
 import installPods from '../../tools/installPods';
 import {processTemplateName} from './templateName';
 import banner from './banner';
 import {getLoader} from '../../tools/loader';
 
 const DEFAULT_VERSION = 'latest';
+const DEVELOPMENT_VERSION = /^(file|http|https|git|git\+ssh|git\+file|github|bitbucket)\:/;
```

Comment:
The `--template` arg of CLI now supports whatever Yarn/npm supports and it's perfectly valid to install template from a local directory or a repository. It's an oversimplification to assume that `DEVELOPMENT_VERSION` is always gonna be `react-native`/RN fork repo. I think we'd need to introduce another flag for this specific use case, if we want to support it automatically.

Have you seen https://github.com/react-native-community/cli/blob/master/CONTRIBUTING.md#running-cli-with-react-native-from-the-source?

Reply:
That regex is only being used for the value of `--version` to differentiate between whether or not you're passing a version or a tag (like `latest`). If you just use `--template` and don't provide a value to `--version` at all, this regex won't be used